raven/senders/lua-resty-http.lua

```diff
@@ -0,0 +1,79 @@
+-- vim: st=4 sts=4 sw=4 et:
+--- Network backend using [lua-resty-http](https://github.com/ledgetech/lua-resty-http).
+--- Supports https, http, and keepalive for better performance.
+--
+-- @module raven.senders.lua-resty-http
+-- @copyright 2014-2017 CloudFlare, Inc.
+-- @license BSD 3-clause (see LICENSE file)
+
+local util = require 'raven.util'
+local http = require 'resty.http'
+local cjson = require 'cjson'
+
+local tostring = tostring
+local setmetatable = setmetatable
+local table_concat = table.concat
+local parse_dsn = util.parse_dsn
+local generate_auth_header = util.generate_auth_header
+local _VERSION = util._VERSION
+local _M = {}
+
+local mt = {}
+mt.__index = mt
+
+function mt:send(json_str)
+    local httpc = http.new()
+    local res, err = httpc:request_uri(self.server, {
```

Comment:
Note that sockets are not always available with `lua_nginx_module`: some phases such as `init`, `set` and `log` don't support async operations. The plain `ngx` module works around this limitation by spawning a timer that does the actual sending.

Perhaps there is a way to extract that logic so it doesn't need to be replicated here.